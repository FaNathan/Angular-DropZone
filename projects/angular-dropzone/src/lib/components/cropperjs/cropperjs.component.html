<div class="cropper-container" [ngClass]="{ roundCropper: avatar.round }" [ngStyle]="{ 'width.px': avatar.width, 'height.px': avatar.height }">
  <img class="image-cropper" #cropper />
</div>
<div class="actions">
  <svg
    (click)="onCancel()"
    class="cancel-file"
    width="30"
    height="30"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9.17218 14.8284L12.0006 12M14.829 9.17157L12.0006 12M12.0006 12L9.17218 9.17157M12.0006 12L14.829 14.8284"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      class="circle-x"
    />
    <path
      class="circle"
      d="M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22Z"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
  <svg
    class="success-file"
    (click)="getImageBlob()"
    width="30"
    height="30"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path d="M7 12.5L10 15.5L17 8.5" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" class="circle-x" />
    <path
      class="circle"
      d="M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22Z"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
