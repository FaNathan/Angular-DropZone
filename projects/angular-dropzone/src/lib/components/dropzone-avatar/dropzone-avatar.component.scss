@import '../variables.scss';
:host {
  height: 100%;
  width: 100%;
  display: flex;
  .dropzone-container {
    color: $secondary-text-color;
    display: grid;
    background-color: $background-color;
    border-radius: 5px;
    box-sizing: border-box;
    transition: height 0.3s ease-out;
    height: 100%;
    width: 100%;
    background: none;
    cursor: pointer;
    &:hover,
    &:focus {
      .actions-container {
        height: 90px !important;
        &.uploading,
        &.upload-error {
          height: 100% !important;
        }
      }
    }
    &.drag-over {
      opacity: 0.5;
    }

    .avatar-uploader-container {
      border: 2px solid #575e65ba;
      color: #575e65ba;
      display: flex;
      align-items: flex-start;
      justify-content: center;
      box-sizing: border-box;
      position: relative;
      .uploaded-image {
        height: 100%;
        width: 100%;
        display: flex;
        .valid-image {
          overflow: hidden;
          object-fit: cover;
        }
        .valid-svg {
          width: 100%;
          height: 100%;
          stroke-width: 0.3;
          circle {
            stroke-width: 0.3;
          }
        }
      }
    }
    .actions-container {
      background-color: #787878b5;
      position: absolute;
      bottom: 0;
      left: 0;
      width: 100%;
      height: 0;
      display: flex;
      align-items: center;
      transition: height 0.3s ease-in-out;
      justify-content: center;
      overflow: hidden;
      flex-direction: column;
      color: white;
      &.upload-error {
        background-color: $error-color;
        height: 100%;
      }
      &.uploading {
        background-color: $action-color;
        height: 100%;
      }
      > svg {
        cursor: pointer;
        opacity: 0.7;
        &:hover {
          opacity: 1;
        }
      }
      &.small-container {
        .action-button {
          .title {
            display: none;
          }
        }
      }
      .action-button {
        display: flex;
        align-items: center;
        flex-direction: column;
        width: 30px;
        height: 45px;
        font-size: 10px;
        margin: 0 5px;
      }
      .error-messages {
        display: flex;
        flex-direction: column;
        padding: 0 3ch;
        overflow: auto;
        font-size: 12px;
      }
      .panel {
        display: flex;
        align-items: center;
        justify-content: space-evenly;
        width: 100%;
        justify-content: center;
        height: 90px;
        .browse-file {
          display: flex;
          height: 100%;
          width: 100%;
          align-items: center;
          justify-content: center;
          flex-direction: column;
        }
      }
    }
  }
}
