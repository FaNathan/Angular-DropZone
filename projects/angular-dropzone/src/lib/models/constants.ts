
export enum FileStatus {
  Ready = 'Ready',
  Pending = 'Pending',
  Uploading = 'Uploading',
  Processing = 'Processing',
  Completed = 'Completed',
  Unsupported = 'Unsupported',
  Canceled = 'Canceled',
}

export const defaultConcurrentUploadLimit = 5;
export const defaultMaxFileLimit = -1; // Unlimited
export const defaultMaxFileSize = 40000;
export const defaultFileSizeUnit = 'MB';
export const defaultChunkUploadSize = 0;

export type FileSizeTypes = "KB" | "MB" | "GB";

export const SizeUnits: { [key: string]: number } = {
  KB: 1024,
  MB: 1048576,
  GB: 1073741824,
};
