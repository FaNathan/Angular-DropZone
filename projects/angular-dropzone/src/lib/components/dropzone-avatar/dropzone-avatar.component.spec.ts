import { ComponentFixture, fakeAsync, flush, flushMicrotasks, TestBed, tick, waitForAsync } from '@angular/core/testing';
import { MockProvider } from 'ng-mocks';
import { filter, interval, map, take, takeWhile, tap } from 'rxjs';
import { AngularDropzoneAPI } from '../../models/file.model';
import { AngularDropzoneService } from '../../services/angular-dropzone.service';
import { based64Image, dumpFiles_6 } from '../../models/dumpData';
import { DropzoneAvatarComponent } from './dropzone-avatar.component';
import { CropperjsComponent } from '../cropperjs/cropperjs.component';
import { Buffer } from 'buffer';
import { By } from '@angular/platform-browser';

describe('DropzoneAvatarComponent', () => {
  let component: DropzoneAvatarComponent;
  let fixture: ComponentFixture<DropzoneAvatarComponent>;
  let dumpFiles: Readonly<FileList>;
  let uploadService = MockProvider<AngularDropzoneService>(AngularDropzoneService, {
    uploadMedia:
      (file: File, uploadAPI: AngularDropzoneAPI) => interval(1).pipe(
        take(21),
        map((id) => id === 20 ? ({ type: 4, loaded: 100 }) : ({ type: 1, loaded: id * 5 })),
      )
  })
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [DropzoneAvatarComponent, CropperjsComponent],
      providers: [uploadService]
    })
      .compileComponents();
    dumpFiles = dumpFiles_6;
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(DropzoneAvatarComponent);
    component = fixture.componentInstance;
    component.uploadAPI = new AngularDropzoneAPI('localhost', 'POST');
    fixture.detectChanges();
  });
  function zero_file() {
    expect(component).toBeTruthy();
    expect(component.files.length).toBe(0);
  }
  function dataUrlToFile(dataUrl: string, filename: string): File | undefined {
    const arr = dataUrl.split(',');
    if (arr.length < 2) { return undefined; }
    const mimeArr = arr[0].match(/:(.*?);/);
    if (!mimeArr || mimeArr.length < 2) { return undefined; }
    const mime = mimeArr[1];
    const buff = Buffer.from(arr[1], 'base64');
    return new File([buff], filename, { type: mime });
  }
  function one_file() {

    const file = dataUrlToFile(based64Image, dumpFiles[0].name);

    component.onBrowseFiles({ target: { files: { length: 1, 0: file } } } as unknown as Event);
  }
  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should go to edit mode on valid file', () => {
    one_file();
    component.cdRef.detectChanges();
    expect(component.avatarEditMode).toBe(true);
    expect(fixture).toMatchSnapshot();

  })
  it(`shouldn't go to edit mode on invalid file name`, () => {
    component.allowedFormats = ["PNG"];
    component.ngOnInit();
    component.onBrowseFiles({ target: { files: { length: 1, 0: { ...dumpFiles[3], name: 'abc.wrongType' } } } } as unknown as Event);
    component.cdRef.detectChanges();
    expect(component.avatarEditMode).toBe(false);
    expect(fixture).toMatchSnapshot();
  })
  it(`shouldn't go to edit mode on valid file Mime type`, () => {
    component.onBrowseFiles({ target: { files: { length: 1, 0: { ...dumpFiles[3], type: 'wrongMimeType' } } } } as unknown as Event);
    component.cdRef.detectChanges();
    expect(component.avatarEditMode).toBe(false);

    expect(fixture).toMatchSnapshot();
  })
  it(`should back to first state on cancel edit`, () => {
    one_file();
    component.cdRef.detectChanges();

    expect(fixture.debugElement.query(By.css('.cancel-file')).nativeElement).toBeTruthy();
    expect(fixture.debugElement.query(By.css('.success-file')).nativeElement).toBeTruthy();
    fixture.debugElement.query(By.css('.cancel-file')).nativeElement.dispatchEvent(new Event('click'));
    expect(component.avatarEditMode).toBe(false);
    expect(component.files.length).toBe(0);
    expect(fixture).toMatchSnapshot();
  })
  it(`should not be able to add file on editmode`, () => {
    one_file();
    component.cdRef.detectChanges();
    one_file();
    component.cdRef.detectChanges();
    one_file();
    component.cdRef.detectChanges();
    expect(component.files.length).toBe(1);
    expect(fixture).toMatchSnapshot();
  })
  it(`should upload after a sccuess crop`, fakeAsync(() => {
    const componentSpy =  jest.spyOn(component,'upload');
    one_file();
    component.cdRef.detectChanges();
    // click on success svg ? 
    component.onAvatarCropped(dataUrlToFile(based64Image, dumpFiles[0].name)!);
    expect(component.files.length).toBe(1);
    tick(800);
    expect(componentSpy).toBeCalledTimes(1);
    expect(fixture).toMatchSnapshot();
  }))

});
