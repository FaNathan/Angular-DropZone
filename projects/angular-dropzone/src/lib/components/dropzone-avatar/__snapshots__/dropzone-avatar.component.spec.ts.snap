// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropzoneAvatarComponent should back to first state on cancel edit 1`] = `
<dropzone-avatar
  __ngContext__={[Function LRootView]}
  allowedFormats={[Function Array]}
  allowedFormatsString={[Function String]}
  autoUpload="false"
  avatar={[Function Object]}
  avatarEditMode="false"
  cdRef={[Function ViewRef$1]}
  chunkUploadSize="0"
  concurrentUploadLimit={[Function Number]}
  container={[Function ElementRef]}
  displayUnit={[Function String]}
  dragEnterCounter="0"
  dropZoneService={[Function AngularDropzoneService]}
  fileRow={[Function QueryList]}
  fileSizeUnit={[Function String]}
  fileStatus={[Function Object]}
  files={[Function Array]}
  keepInvalidFiles={[Function Boolean]}
  maxFileLimit={[Function Number]}
  maxFileSize={[Function Number]}
  multiple={[Function Boolean]}
  uploadAPI={[Function AngularDropzoneAPI]}
  uploaded={[Function EventEmitter_]}
  validateFunctions={[Function Array]}
>
  <div
    class="dropzone-container"
  >
    <input
      accept="image/*,"
      id="inputFile"
      style="display: none;"
      type="file"
    />
    <div
      class="avatar-uploader-container"
      style="width: 150px; height: 150px; border-radius: 50%; overflow: visible;"
    >
      <lib-cropperjs>
        <div
          class="cropper-container roundCropper"
          style="width: 150px; height: 150px;"
        >
          <img
            class="image-cropper"
          />
        </div>
        <div
          class="actions"
        >
          <svg
            class="cancel-file"
            fill="none"
            height="30"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="circle-x"
              d="M9.17218 14.8284L12.0006 12M14.829 9.17157L12.0006 12M12.0006 12L9.17218 9.17157M12.0006 12L14.829 14.8284"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
            <path
              class="circle"
              d="M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22Z"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <svg
            class="success-file"
            fill="none"
            height="30"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="circle-x"
              d="M7 12.5L10 15.5L17 8.5"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
            <path
              class="circle"
              d="M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22Z"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
      </lib-cropperjs>
    </div>
  </div>
</dropzone-avatar>
`;

exports[`DropzoneAvatarComponent should create 1`] = `
<dropzone-avatar
  __ngContext__={[Function LRootView]}
  allowedFormats={[Function Array]}
  allowedFormatsString={[Function String]}
  autoUpload="false"
  avatar={[Function Object]}
  avatarEditMode="false"
  cdRef={[Function ViewRef$1]}
  chunkUploadSize="0"
  concurrentUploadLimit={[Function Number]}
  container={[Function ElementRef]}
  displayUnit={[Function String]}
  dragEnterCounter="0"
  dropZoneService={[Function AngularDropzoneService]}
  fileRow={[Function QueryList]}
  fileSizeUnit={[Function String]}
  fileStatus={[Function Object]}
  files={[Function Array]}
  keepInvalidFiles={[Function Boolean]}
  maxFileLimit={[Function Number]}
  maxFileSize={[Function Number]}
  multiple={[Function Boolean]}
  uploadAPI={[Function AngularDropzoneAPI]}
  uploaded={[Function EventEmitter_]}
  validateFunctions={[Function Array]}
>
  <div
    class="dropzone-container"
  >
    <input
      accept="image/*,"
      id="inputFile"
      style="display: none;"
      type="file"
    />
    <div
      class="avatar-uploader-container"
      style="width: 150px; height: 150px; border-radius: 50%; overflow: hidden;"
    >
      <div
        class="uploaded-image"
      >
        <svg
          class="valid-svg"
          fill="none"
          height="24"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 18V17C7 14.2386 9.23858 12 12 12V12C14.7614 12 17 14.2386 17 17V18"
            stroke="currentColor"
            stroke-linecap="round"
          />
          <path
            d="M12 12C13.6569 12 15 10.6569 15 9C15 7.34315 13.6569 6 12 6C10.3431 6 9 7.34315 9 9C9 10.6569 10.3431 12 12 12Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <div
        class="actions-container small-container"
      >
        <div
          class="panel"
        >
          <div
            class="action-button"
            title="Add"
          >
            <svg
              fill="none"
              height="30"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 21L12 21H21"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
              <path
                d="M12.2218 5.82839L15.0503 2.99996L20 7.94971L17.1716 10.7781M12.2218 5.82839L6.61522 11.435C6.42769 11.6225 6.32233 11.8769 6.32233 12.1421L6.32233 16.6776L10.8579 16.6776C11.1231 16.6776 11.3774 16.5723 11.565 16.3847L17.1716 10.7781M12.2218 5.82839L17.1716 10.7781"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="title"
            >
              Add
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</dropzone-avatar>
`;

exports[`DropzoneAvatarComponent should go to edit mode on valid file 1`] = `
<dropzone-avatar
  __ngContext__={[Function LRootView]}
  allowedFormats={[Function Array]}
  allowedFormatsString={[Function String]}
  autoUpload="false"
  avatar={[Function Object]}
  avatarEditMode={[Function Boolean]}
  cdRef={[Function ViewRef$1]}
  chunkUploadSize="0"
  concurrentUploadLimit="0"
  container={[Function ElementRef]}
  displayUnit={[Function String]}
  dragEnterCounter="0"
  dropZoneService={[Function AngularDropzoneService]}
  fileRow={[Function QueryList]}
  fileSizeUnit={[Function String]}
  fileStatus={[Function Object]}
  files={[Function Array]}
  keepInvalidFiles={[Function Boolean]}
  maxFileLimit={[Function Number]}
  maxFileSize={[Function Number]}
  multiple={[Function Boolean]}
  uploadAPI={[Function AngularDropzoneAPI]}
  uploaded={[Function EventEmitter_]}
  validateFunctions={[Function Array]}
>
  <div
    class="dropzone-container"
  >
    <input
      accept="image/*,"
      id="inputFile"
      style="display: none;"
      type="file"
    />
    <div
      class="avatar-uploader-container"
      style="width: 150px; height: 150px; border-radius: 50%; overflow: visible;"
    >
      <lib-cropperjs>
        <div
          class="cropper-container roundCropper"
          style="width: 150px; height: 150px;"
        >
          <img
            class="image-cropper"
          />
        </div>
        <div
          class="actions"
        >
          <svg
            class="cancel-file"
            fill="none"
            height="30"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="circle-x"
              d="M9.17218 14.8284L12.0006 12M14.829 9.17157L12.0006 12M12.0006 12L9.17218 9.17157M12.0006 12L14.829 14.8284"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
            <path
              class="circle"
              d="M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22Z"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <svg
            class="success-file"
            fill="none"
            height="30"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="circle-x"
              d="M7 12.5L10 15.5L17 8.5"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
            <path
              class="circle"
              d="M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22Z"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
      </lib-cropperjs>
    </div>
  </div>
</dropzone-avatar>
`;

exports[`DropzoneAvatarComponent should not be able to add file on editmode 1`] = `
<dropzone-avatar
  __ngContext__={[Function LRootView]}
  allowedFormats={[Function Array]}
  allowedFormatsString={[Function String]}
  autoUpload="false"
  avatar={[Function Object]}
  avatarEditMode={[Function Boolean]}
  cdRef={[Function ViewRef$1]}
  chunkUploadSize="0"
  concurrentUploadLimit="0"
  container={[Function ElementRef]}
  displayUnit={[Function String]}
  dragEnterCounter="0"
  dropZoneService={[Function AngularDropzoneService]}
  fileRow={[Function QueryList]}
  fileSizeUnit={[Function String]}
  fileStatus={[Function Object]}
  files={[Function Array]}
  keepInvalidFiles={[Function Boolean]}
  maxFileLimit={[Function Number]}
  maxFileSize={[Function Number]}
  multiple={[Function Boolean]}
  uploadAPI={[Function AngularDropzoneAPI]}
  uploaded={[Function EventEmitter_]}
  validateFunctions={[Function Array]}
>
  <div
    class="dropzone-container"
  >
    <input
      accept="image/*,"
      id="inputFile"
      style="display: none;"
      type="file"
    />
    <div
      class="avatar-uploader-container"
      style="width: 150px; height: 150px; border-radius: 50%; overflow: visible;"
    >
      <lib-cropperjs>
        <div
          class="cropper-container roundCropper"
          style="width: 150px; height: 150px;"
        >
          <img
            class="image-cropper"
          />
        </div>
        <div
          class="actions"
        >
          <svg
            class="cancel-file"
            fill="none"
            height="30"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="circle-x"
              d="M9.17218 14.8284L12.0006 12M14.829 9.17157L12.0006 12M12.0006 12L9.17218 9.17157M12.0006 12L14.829 14.8284"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
            <path
              class="circle"
              d="M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22Z"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <svg
            class="success-file"
            fill="none"
            height="30"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="circle-x"
              d="M7 12.5L10 15.5L17 8.5"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
            <path
              class="circle"
              d="M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22Z"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
      </lib-cropperjs>
    </div>
  </div>
</dropzone-avatar>
`;

exports[`DropzoneAvatarComponent should upload after a sccuess crop 1`] = `
<dropzone-avatar
  __ngContext__={[Function LRootView]}
  allowedFormats={[Function Array]}
  allowedFormatsString={[Function String]}
  autoUpload="false"
  avatar={[Function Object]}
  avatarEditMode="false"
  cdRef={[Function ViewRef$1]}
  chunkUploadSize="0"
  concurrentUploadLimit={[Function Number]}
  container={[Function ElementRef]}
  displayUnit={[Function String]}
  dragEnterCounter="0"
  dropZoneService={[Function AngularDropzoneService]}
  fileRow={[Function QueryList]}
  fileSizeUnit={[Function String]}
  fileStatus={[Function Object]}
  files={[Function Array]}
  keepInvalidFiles={[Function Boolean]}
  maxFileLimit={[Function Number]}
  maxFileSize={[Function Number]}
  multiple={[Function Boolean]}
  upload={[Function Function]}
  uploadAPI={[Function AngularDropzoneAPI]}
  uploaded={[Function EventEmitter_]}
  validateFunctions={[Function Array]}
>
  <div
    class="dropzone-container"
  >
    <input
      accept="image/*,"
      id="inputFile"
      style="display: none;"
      type="file"
    />
    <div
      class="avatar-uploader-container"
      style="width: 150px; height: 150px; border-radius: 50%; overflow: hidden;"
    >
      <div
        class="uploaded-image"
      >
        <img
          class="valid-image"
          draggable="false"
          src=""
          style="border-radius: 50%;"
        />
      </div>
      <div
        class="actions-container small-container"
      >
        <div
          class="panel"
        >
          <div
            class="action-button"
            title="Change"
          >
            <svg
              fill="none"
              height="30"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 21L12 21H21"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
              <path
                d="M12.2218 5.82839L15.0503 2.99996L20 7.94971L17.1716 10.7781M12.2218 5.82839L6.61522 11.435C6.42769 11.6225 6.32233 11.8769 6.32233 12.1421L6.32233 16.6776L10.8579 16.6776C11.1231 16.6776 11.3774 16.5723 11.565 16.3847L17.1716 10.7781M12.2218 5.82839L17.1716 10.7781"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="title"
            >
              Change
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</dropzone-avatar>
`;

exports[`DropzoneAvatarComponent shouldn't go to edit mode on invalid file name 1`] = `
<dropzone-avatar
  __ngContext__={[Function LRootView]}
  allowedFormats={[Function Array]}
  allowedFormatsString={[Function String]}
  autoUpload="false"
  avatar={[Function Object]}
  avatarEditMode="false"
  cdRef={[Function ViewRef$1]}
  chunkUploadSize="0"
  concurrentUploadLimit={[Function Number]}
  container={[Function ElementRef]}
  displayUnit={[Function String]}
  dragEnterCounter="0"
  dropZoneService={[Function AngularDropzoneService]}
  fileRow={[Function QueryList]}
  fileSizeUnit={[Function String]}
  fileStatus={[Function Object]}
  files={[Function Array]}
  keepInvalidFiles={[Function Boolean]}
  maxFileLimit={[Function Number]}
  maxFileSize={[Function Number]}
  multiple={[Function Boolean]}
  uploadAPI={[Function AngularDropzoneAPI]}
  uploaded={[Function EventEmitter_]}
  validateFunctions={[Function Array]}
>
  <div
    class="dropzone-container"
  >
    <input
      accept="image/*,.png,"
      id="inputFile"
      style="display: none;"
      type="file"
    />
    <div
      class="avatar-uploader-container"
      style="width: 150px; height: 150px; border-radius: 50%; overflow: hidden;"
    >
      <div
        class="uploaded-image"
      >
        <svg
          class="valid-svg"
          fill="none"
          height="24"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 18V17C7 14.2386 9.23858 12 12 12V12C14.7614 12 17 14.2386 17 17V18"
            stroke="currentColor"
            stroke-linecap="round"
          />
          <path
            d="M12 12C13.6569 12 15 10.6569 15 9C15 7.34315 13.6569 6 12 6C10.3431 6 9 7.34315 9 9C9 10.6569 10.3431 12 12 12Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <div
        class="actions-container upload-error small-container"
      >
        <div
          class="error-container"
        >
          <span
            class="error-title"
          >
            Incorrect File!
          </span>
          <div
            class="error-messages"
          >
            <span>
              - File type is not supported.
            </span>
            <span>
              - File type is not supported.
            </span>
          </div>
        </div>
        <div
          class="panel"
        >
          <div
            class="action-button"
            title="Change"
          >
            <svg
              fill="none"
              height="30"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 21L12 21H21"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
              <path
                d="M12.2218 5.82839L15.0503 2.99996L20 7.94971L17.1716 10.7781M12.2218 5.82839L6.61522 11.435C6.42769 11.6225 6.32233 11.8769 6.32233 12.1421L6.32233 16.6776L10.8579 16.6776C11.1231 16.6776 11.3774 16.5723 11.565 16.3847L17.1716 10.7781M12.2218 5.82839L17.1716 10.7781"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="title"
            >
              Change
            </div>
          </div>
          <div
            class="action-button"
            title="Cancel"
          >
            <svg
              fill="none"
              height="30"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="circle-x"
                d="M9.17218 14.8284L12.0006 12M14.829 9.17157L12.0006 12M12.0006 12L9.17218 9.17157M12.0006 12L14.829 14.8284"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
              <path
                class="circle"
                d="M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22Z"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="title"
            >
              Cancel
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</dropzone-avatar>
`;

exports[`DropzoneAvatarComponent shouldn't go to edit mode on valid file Mime type 1`] = `
<dropzone-avatar
  __ngContext__={[Function LRootView]}
  allowedFormats={[Function Array]}
  allowedFormatsString={[Function String]}
  autoUpload="false"
  avatar={[Function Object]}
  avatarEditMode="false"
  cdRef={[Function ViewRef$1]}
  chunkUploadSize="0"
  concurrentUploadLimit={[Function Number]}
  container={[Function ElementRef]}
  displayUnit={[Function String]}
  dragEnterCounter="0"
  dropZoneService={[Function AngularDropzoneService]}
  fileRow={[Function QueryList]}
  fileSizeUnit={[Function String]}
  fileStatus={[Function Object]}
  files={[Function Array]}
  keepInvalidFiles={[Function Boolean]}
  maxFileLimit={[Function Number]}
  maxFileSize={[Function Number]}
  multiple={[Function Boolean]}
  uploadAPI={[Function AngularDropzoneAPI]}
  uploaded={[Function EventEmitter_]}
  validateFunctions={[Function Array]}
>
  <div
    class="dropzone-container"
  >
    <input
      accept="image/*,"
      id="inputFile"
      style="display: none;"
      type="file"
    />
    <div
      class="avatar-uploader-container"
      style="width: 150px; height: 150px; border-radius: 50%; overflow: hidden;"
    >
      <div
        class="uploaded-image"
      >
        <svg
          class="valid-svg"
          fill="none"
          height="24"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 18V17C7 14.2386 9.23858 12 12 12V12C14.7614 12 17 14.2386 17 17V18"
            stroke="currentColor"
            stroke-linecap="round"
          />
          <path
            d="M12 12C13.6569 12 15 10.6569 15 9C15 7.34315 13.6569 6 12 6C10.3431 6 9 7.34315 9 9C9 10.6569 10.3431 12 12 12Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <div
        class="actions-container upload-error small-container"
      >
        <div
          class="error-container"
        >
          <span
            class="error-title"
          >
            Incorrect File!
          </span>
          <div
            class="error-messages"
          >
            <span>
              - File type is not supported.
            </span>
          </div>
        </div>
        <div
          class="panel"
        >
          <div
            class="action-button"
            title="Change"
          >
            <svg
              fill="none"
              height="30"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 21L12 21H21"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
              <path
                d="M12.2218 5.82839L15.0503 2.99996L20 7.94971L17.1716 10.7781M12.2218 5.82839L6.61522 11.435C6.42769 11.6225 6.32233 11.8769 6.32233 12.1421L6.32233 16.6776L10.8579 16.6776C11.1231 16.6776 11.3774 16.5723 11.565 16.3847L17.1716 10.7781M12.2218 5.82839L17.1716 10.7781"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="title"
            >
              Change
            </div>
          </div>
          <div
            class="action-button"
            title="Cancel"
          >
            <svg
              fill="none"
              height="30"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="circle-x"
                d="M9.17218 14.8284L12.0006 12M14.829 9.17157L12.0006 12M12.0006 12L9.17218 9.17157M12.0006 12L14.829 14.8284"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
              <path
                class="circle"
                d="M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22Z"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            <div
              class="title"
            >
              Cancel
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</dropzone-avatar>
`;
